.clang-format

```diff
@@ -0,0 +1,25 @@
+SpaceAfterTemplateKeyword: 'false'
+AllowShortBlocksOnASingleLine: 'true'
+AllowShortFunctionsOnASingleLine: All
+AlwaysBreakBeforeMultilineStrings: 'true'
+AlwaysBreakTemplateDeclarations: 'true'
+BreakBeforeBraces: Allman
+SpaceBeforeCpp11BracedList: 'true'
+ColumnLimit: '120'
```

Comment:
Really not sure about this - we could just remove it

Reply:
for code convention we need it

Reply:
And this should be merged before anything else